Pool/Authenticate.vb

```diff
@@ -0,0 +1,275 @@
+        Try
+            hashpwdFromDB = pwdHashDS.Tables(0).Rows(0).Item("password")
+            hashFormatSplit = hashpwdFromDB.Split(":")
+        Catch ex As Exception
+            Dim logHashSplitFail As New Logging(Now, "Hash Split Failed", ex.ToString)
+            logHashSplitFail.LogAction()
+            logHashSplitFail.LogActionEmail()
+            Return False
+        End Try
+
+        Dim lastTimeupdated As New DateTime
+        Try
+            DateTime.TryParse(pwdHashDS.Tables(0).Rows(0).Item("LastUpdated"), lastTimeupdated)
+        Catch ex As Exception
+            Debug.WriteLine("first time password")
+        End Try
+
+        'Split to get length:salt:hash
+        Dim hashfromDB As New HashFormat
+        Try
+            With hashfromDB
+                .length = hashFormatSplit.First
+                .salt = Convert.FromBase64String(hashFormatSplit(1))
+                .hash = Convert.FromBase64String(hashFormatSplit.Last)
+            End With
+        Catch ex As Exception
+            Dim logHashSplitFail As New Logging(Now, "Hash Split Failed", ex.ToString)
+            logHashSplitFail.LogAction()
+            logHashSplitFail.LogActionEmail()
+            Return False
+        End Try
+
+        'Generate hash from passed password      
+        Dim hashBytefromForm = CalculateHash(hashfromDB.salt, Password)
+        'Compare until we find an incorrect byte
+        'If we make it to the end, then it's good
+        isLoginCreds = CompareHash(hashBytefromForm, hashfromDB.hash)
+
+        'If their login, then put a timestamp
+        If isLoginCreds.Equals(True) Then
+            UserMod.Password = hashpwdfromDB
+            Dim insertLoginSQl As New StringBuilder
+            With insertLoginSQl
+                .Append("exec [insLoginUser] ")
+                .Append($"@user = '{User}',")
+                .Append($"@lastLogin = '{Now.ToString}',")
+                .Append($"@loginDevice = '{Environment.MachineName.ToString}',")
+                .Append($"@hashMatch = {Convert.ToInt32(isLoginCreds)},")
+                .Append($"@Success = {Convert.ToInt32(isLoginCreds)}")
+            End With
+            CheckLastUpdated(lastTimeupdated)
+            Debug.WriteLine(hashedPasswordDS.GetAllResults(insertLoginSQl.ToString))
+            SetUser(hashedPasswordDS, isLoginCreds)
+        End If
+        Debug.WriteLine(isLoginCreds)
+        Return isLoginCreds
+    End Function
+
+    ''' <summary>
+    ''' Check if it's been 6 months or more
+    ''' Update Password
+    ''' </summary>
+    ''' <param name="lastimeUpdated"></param>
+    Private Sub CheckLastUpdated(ByVal lastimeUpdated)
+        If DateDiff(DateInterval.Month, lastimeUpdated, Now) >= 6 Then
+            Dim emailChangePassword As New Email(New String() {User}.ToList)
+            Dim reminderSent As Boolean = emailChangePassword.SendPasswordReminder(User, lastimeUpdated)
+        End If
+    End Sub
```

Comment:
Check this to make sure an email is not sent automatically for new users. If it needs a check if it's a valid one, we can save that for later.

Reply:
Just need it added in the insNewUser sp. It needed a date to skip that email.